apps/job-service/src/jobs/delayed-jobs.controller.ts

```diff
@@ -97,19 +97,23 @@ export class DelayedJobsController {
       },
     },
   })
+  @JsonApiResponse({ data: { type: DelayedJobDto } })
   @ApiException(() => UnauthorizedException, { description: 'Authentication failed.' })
   @ApiException(() => BadRequestException, { description: 'Invalid payload or IDs provided.' })
   @ApiException(() => NotFoundException, { description: 'One or more jobs specified in the payload could not be found.' })
   async bulkClearJobs(
-    @Body() bulkClearJobsDto: JobBulkUpdateBodyDto,
+    @Body() bulkClearJobsDto: DelayedJobBulkUpdateBodyDto,
     @Request() { authenticatedUserId }
-  ): Promise<{ data: JobData[] }> {
+  ): Promise<JsonApiDocument> {
     const jobUpdates = bulkClearJobsDto.data;
-
+  
     if (!jobUpdates || jobUpdates.length === 0) {
       throw new BadRequestException('No jobs provided in the payload.');
     }
```

Comment:
This should be handled by validation decorators in the request DTO. See the bulk update polygon indicator endpoint for examples. In general, we have tools in place to limit the amount of manual checking needed within controller methods. Ideally, by the time the request makes it to the controller, the inputs have been more or less validated and don't need further checking.

Reply:
removed too